src/state_machine.zig

```diff
@@ -1015,11 +998,9 @@ pub fn StateMachineType(
 
             var scan_lookup_buffer = std.mem.bytesAsSlice(
                 Transfer,
-                self.scan_lookup_buffer[0 .. @sizeOf(Transfer) *
-                    // We must be constrained to the same limit as `create_transfers`.
-                    constants.batch_max.create_transfers],
+                // We must be constrained to the same limit as `create_transfers`.
+                self.scan_lookup_buffer[0..self.scan_buffer_size_create_transfers],
```

Comment:
Hm, what's the reason for this constrained? If we have a small `request_size_limit`, what would break if a scan returns a large batch? Is it just the memory for scan buffers themselves?

Reply:
Great catch; it doesn't need to be constrained! I was getting this mixed up with `lookup_{accounts,transfers}`, where the output size corresponds to the input size.

Reply:
Ah, wait, pulses do need to be constrained! See: https://github.com/tigerbeetle/tigerbeetle/pull/1981/commits/2d06376abebf42deb87fbd06e41777f512e31bf2